src/devices/tuya.ts

```diff
@@ -18617,4 +18617,235 @@ export const definitions: DefinitionWithExtend[] = [
+                        timing_switch_on: 21,
+                        timing_switch_off: 22,
+                    }),
+                ],
+                [
+                    102,
+                    "over_voltage_setting",
+                    tuya.valueConverterBasic.lookup({
+                        Ignore: tuya.enum(0),
+                        Alarm: tuya.enum(1),
+                        Trip: tuya.enum(2),
+                    }),
+                ],
+                [
+                    103,
+                    "under_voltage_setting",
+                    tuya.valueConverterBasic.lookup({
+                        Ignore: tuya.enum(0),
+                        Alarm: tuya.enum(1),
+                        Trip: tuya.enum(2),
+                    }),
+                ],
+                [
+                    104,
+                    "over_current_setting",
+                    tuya.valueConverterBasic.lookup({
+                        Ignore: tuya.enum(0),
+                        Alarm: tuya.enum(1),
+                        Trip: tuya.enum(2),
+                    }),
+                ],
+                [
+                    105,
+                    "over_power_setting",
+                    tuya.valueConverterBasic.lookup({
+                        Ignore: tuya.enum(0),
+                        Alarm: tuya.enum(1),
+                        Trip: tuya.enum(2),
+                    }),
+                ],
+                [
+                    107,
+                    "temperature_setting",
+                    tuya.valueConverterBasic.lookup({
+                        Ignore: tuya.enum(0),
+                        Alarm: tuya.enum(1),
+                        Trip: tuya.enum(2),
+                    }),
+                ],
+                [
+                    108,
+                    "leakage_setting",
+                    tuya.valueConverterBasic.lookup({
+                        Ignore: tuya.enum(0),
+                        Alarm: tuya.enum(1),
+                        Trip: tuya.enum(2),
+                    }),
+                ],
+                [112, "auto_reclosing", tuya.valueConverter.onOff],
+                [113, "restore_default", tuya.valueConverter.onOff],
+                [114, "over_current_threshold", tuya.valueConverter.raw],
+                [115, "over_voltage_threshold", tuya.valueConverter.raw],
+                [116, "under_voltage_threshold", tuya.valueConverter.raw],
+                [117, "leakage_threshold", tuya.valueConverter.raw],
+                [118, "temperature_threshold", tuya.valueConverter.divideBy10],
+                [119, "over_power_threshold", tuya.valueConverter.raw],
+                [131, "temperature", tuya.valueConverter.divideBy10],
+                [143, "overcurrent_recloser", tuya.valueConverter.onOff],
+                [144, "leakage_recloser", tuya.valueConverter.onOff],
+                [145, "overpower_recloser", tuya.valueConverter.onOff],
```

Comment:
Thanks for the PR! I see the `TOWSMR1` definition is very similar except datapoint 143, 144 and 145, did you confirm those work and do you think the TOWSMR1 might also support those?